Basic-Car-Maintenance/Shared/Dashboard/Views/CarMaintenancePDFGenerator.swift

```diff
@@ -0,0 +1,230 @@
+        context: UIGraphicsPDFRendererContext,
+        yPosition: inout CGFloat
+    ) {
+        let titleAttributes: [NSAttributedString.Key: Any] = [
+            .font: UIFont.boldSystemFont(ofSize: 20),
+            .foregroundColor: UIColor.black
+        ]
+        let subtitleAttributes: [NSAttributedString.Key: Any] = [
+            .font: UIFont.systemFont(ofSize: 16),
+            .foregroundColor: UIColor.black
+        ]
+        
+        let titleString = "Basic Car Maintenance"
+        let eventTitleString = "Maintenance Events"
+        guard 
+            let startDate = events.first?.date.formatted(date: .numeric, time: .omitted),
+            let endDate = events.last?.date.formatted(date: .numeric, time: .omitted)
+        else { return }
+        let dateRangeString = "From \(startDate) to \(endDate)"
+        
+        let titleSize = titleString.size(withAttributes: titleAttributes)
+        let vehicleSize = vehicleName.size(withAttributes: subtitleAttributes)
+        let eventTitleSize = eventTitleString.size(withAttributes: subtitleAttributes)
+        let dateRangeSize = dateRangeString.size(withAttributes: subtitleAttributes)
+        
+        // Center the headers
+        let titleX = (pageWidth - titleSize.width) / 2
+        let vehicleX = (pageWidth - vehicleSize.width) / 2
+        let eventTitleX = (pageWidth - eventTitleSize.width) / 2
+        let dateRangeX = (pageWidth - dateRangeSize.width) / 2
+        
+        // Draw the main title centered
+        titleString.draw(at: CGPoint(x: titleX, y: yPosition), withAttributes: titleAttributes)
+        yPosition += 30
+        
+        // Draw the vehicle name centered
+        vehicleName.draw(at: CGPoint(x: vehicleX, y: yPosition), withAttributes: subtitleAttributes)
+        yPosition += 30
+        
+        // Draw the maintenance event title centered
+        eventTitleString.draw(at: CGPoint(x: eventTitleX, y: yPosition), withAttributes: subtitleAttributes)
+        yPosition += 30
+        
+        // Draw date range centered
+        dateRangeString.draw(at: CGPoint(x: dateRangeX, y: yPosition), withAttributes: subtitleAttributes)
+        yPosition += 50  // Add more space after the headers
+        
+        drawColumnsHeaders(yPosition: &yPosition)
+    }
+    
+    private func drawColumnsHeaders(
+        yPosition: inout CGFloat
+    ) {
+        // Draw the headers of each column
+        let dateColumnHeader = "Date"
+        let vehicleColumnHeader = "Vehicle Name"
+        let noteColumnHeader = "Notes"
+        
+        let subtitleAttributes: [NSAttributedString.Key: Any] = [
+            .font: UIFont.boldSystemFont(ofSize: 16),
+            .foregroundColor: UIColor.black
+        ]
+        
+        dateColumnHeader
+            .draw(
+                at: CGPoint(
+                    x: leftMargin,
+                    y: yPosition
+                ),
+                withAttributes: subtitleAttributes
+            )
```

Comment:
```suggestion
        dateColumnHeader
            .draw(at: CGPoint(x: leftMargin, y: yPosition),
                  withAttributes: subtitleAttributes
            )
```

can you reformat it like this